app/views/main_content_panel.py

```diff
@@ -2168,8 +2168,14 @@ def _check_git_repos_for_update(self, repo_paths: list[str]) -> None:
 
                         # Get the local and remote refs
                         local_ref = repo.head.reference
-                        refs = repo.refs()
-                        remote_ref = refs[f"origin/{local_ref.name}"]
+                        remote_ref = next(
+                            (
+                                ref
+                                for ref in repo.refs
+                                if f"origin/{local_ref.name}" in ref.path
+                            ),
+                            None,
+                        )
 
                         # Check if the local branch is behind the remote branch
                         if local_ref.commit != remote_ref.commit:
```

Comment:
Why was the logging removed from here? I think it'd be useful to know when `remote_ref` is `None`.

Also, when `remote_ref` is `None`, the conditional `if local_ref.commit != remote_ref.commit:` will fail because `None` has no attribute `.commit`. I think you have to check the existence of `remote_ref` first, e.g. `if remote_ref is not None`.

Reply:
hopefully i did not miss anything now